tests/integration/targets/aci_interface_policy_link_level/tasks/main.yml

```
@@ -138,6 +138,45 @@
       - cm_add_policy_again_no_descr.proposed.fabricHIfPol.attributes.name == nm_add_policy_again_no_descr.proposed.fabricHIfPol.attributes.name == 'ansible_test'
       - cm_add_policy_again_no_descr.sent == nm_add_policy_again_no_descr.sent == {}
       - cm_add_policy_again_no_descr.previous.0.fabricHIfPol.attributes.dn== nm_add_policy_again_no_descr.previous.0.fabricHIfPol.attributes.dn == cm_add_policy_again_no_descr.current.0.fabricHIfPol.attributes.dn == nm_add_policy_again_no_descr.current.0.fabricHIfPol.attributes.dn == 'uni/infra/hintfpol-ansible_test'
+  
+  # CHANGE SPEED ON LINK LEVEL POLICY
+  - name: Test each speed setting for Link Level Policy
+    block:
+      - name: Add Link Level Policy with various speeds (check mode)
+        cisco.aci.aci_interface_policy_link_level:
+          <<: *interface_policy_link_level_present
+          speed: "{{ item }}"
+        check_mode: true
+        loop: ["100M", "1G", "10G", "25G", "40G", "50G", "100G", "200G", "400G", "auto"]
+        register: cm_add_policy_speed
+        loop_control:
+          label: "speed-{{ item }}"
+
+      - name: Add Link Level Policy with various speeds (normal mode)
+        cisco.aci.aci_interface_policy_link_level:
+          <<: *interface_policy_link_level_present
+          speed: "{{ item }}"
+        loop: ["100M", "1G", "10G", "25G", "40G", "50G", "100G", "200G", "400G", "auto", "inherit"]
+        register: nm_add_policy_speed
+        loop_control:
+          label: "speed-{{ item }}"
+    
+      - name: Verify each speed setting (normal mode)
+        ansible.builtin.assert:
+          that:
+            - "item.1.current.0.fabricHIfPol.attributes.speed == item.0"
+          quiet: true
+        loop: "{{ ['100M', '1G', '10G', '25G', '40G', '50G', '100G', '200G', '400G', 'auto', 'inherit'] | zip(nm_add_policy_speed.results) }}"
+        loop_control:
+          label: "speed-{{ item.0 }}"
+
+      - name: Verify each speed setting (check mode)
+        ansible.builtin.assert:
+          that:
+            - item is changed
+        loop: "{{ cm_add_policy_speed.results }}"
+        loop_control:
+          label: "speed-{{ item }}"
```

Could you add more specific asserts?

```suggestion
      - name: Verify each speed setting (check mode)
        ansible.builtin.assert:
          that:
            - "item.1.current.0.fabricHIfPol.attributes.speed == 'inherit'"
            - "item.1.proposed.fabricHIfPol.attributes.speed == item.0"
        loop: "{{ ['100M', '1G', '10G', '25G', '40G', '50G', '100G', '200G', '400G', 'auto', 'inherit'] | zip(cm_add_policy_speed.results) }}"
        loop_control:
          label: "speed-{{ item.0 }}"
```